### PR TITLE
#164: Added validation for changelog.md file

### LINKED ISSUE #164: Validate changelog.md
Validate the `changelog.md`

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #164

### CODE REVIEW

File: src/main/java/com/exasol/projectkeeper/validators/changelog/ChangelogFileGenerator.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+     * Generate the content for the changelog file.
+     * 
+     * @param versions list of project versions
+     * @return content for the changelog file
+     */
+    String generate(final List<String> versions) {
+        final List<String> sortedVersions = versions.stream().map(version -> new Semver(version, LOOSE))
+                .sorted(Comparator.reverseOrder()).map(Semver::getValue).collect(Collectors.toList());
+        final StringBuilder templateBuilder = new StringBuilder();
+        templateBuilder.append("# Changes" + NL + NL);
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/projectkeeper/validators/changelog/ChangelogFileGenerator.java:27]: In some project, we start from some later version. For example, we start using changelog from 3.0.0. Would it be a problem if versions below 3.0.0 are missing in the changelog file?

File: src/main/java/com/exasol/projectkeeper/validators/changelog/ChangelogFileValidator.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    private final Path projectDirectory;
+
+    /**
+     * Create a new instance of {@link ChangelogFileValidator}.
+     * 
+     * @param projectDirectory project's root directory
+     */
+    public ChangelogFileValidator(final Path projectDirectory) {
+        super(projectDirectory, Path.of("doc/changes/changelog.md"));
+        this.projectDirectory = projectDirectory;
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/projectkeeper/validators/changelog/ChangelogFileValidator.java:23]: If you also send this to superclass, maybe you just save it in the superclass and use as a protected variable?

File: src/main/java/com/exasol/projectkeeper/validators/changelog/ChangelogFileGenerator.java (hunk 1 above)
Comment (jakobbraun) [reply src/main/java/com/exasol/projectkeeper/validators/changelog/ChangelogFileGenerator.java:27]: This validator requires that all changes_x.x.x.md files that exist are listed in the changelog.md file. So if we did not have changes files for those earlier versions it's not a problem. If we had, we can also list them.

File: src/main/java/com/exasol/projectkeeper/validators/VersionLister.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    public VersionLister(final Path projectDirectory) {
+        this.projectDirectory = projectDirectory;
+    }
+
+    /**
+     * List all project-versions by scanning the doc/changes/ folder.
+     * 
+     * @return list of project versions
+     */
+    public List<String> listVersions() {
```
[the remark this answers is not in the document]
Comment (jakobbraun) [reply src/main/java/com/exasol/projectkeeper/validators/VersionLister.java:32]: I don't like `get` for methods that do something. Get should in my oppinion be reserved for simple getters.
`collect` sounds good.

File: src/main/java/com/exasol/projectkeeper/validators/VersionLister.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        try (final Stream<Path> filesStream = Files.walk(this.projectDirectory.resolve(Path.of("doc", "changes")))) {
+            return filesStream.map(path -> path.getFileName().toString())
+                    .filter(fileName -> fileName.startsWith("changes_"))
+                    .map(fileName -> fileName.replace("changes_", "").replace(".md", "")).collect(Collectors.toList());
```
[the remark this answers is not in the document]
Comment (jakobbraun) [reply src/main/java/com/exasol/projectkeeper/validators/VersionLister.java:36]: Yea, performance doesn't matter here so I chose performance.

File: src/main/java/com/exasol/projectkeeper/validators/changelog/ChangelogFileValidator.java (hunk 2 above)
Comment (jakobbraun) [reply src/main/java/com/exasol/projectkeeper/validators/changelog/ChangelogFileValidator.java:23]: in the super class I don't save it. So I think it's better to save it here.

**Review by AnastasiiaSergienko — APPROVED** [review]

**Pull request merged**